geostore/check_stac_metadata/utils.py

```diff
@@ -251,6 +277,19 @@ def validate(self, url: str) -> None:  # pylint: disable=too-complex
             if next_url not in self.traversed_urls:
                 self.validate(next_url)
 
+    def get_s3_url_as_object_json(self, url: str, s3_response: GeostoreS3Response) -> JsonObject:
+        try:
+            object_json: JsonObject = load(
+                s3_response.response,
+                object_pairs_hook=self.duplicate_object_names_report_builder(url),
+            )
+        except JSONDecodeError as error:
+            self.validation_result_factory.save(
+                url, Check.JSON_PARSE, ValidationResult.FAILED, details={MESSAGE_KEY: str(error)}
+            )
+            raise
+        return object_json
```

Comment:
The `return` can just be in the `try` block - it doesn't change the semantics of the code.

Reply:
Updated. Thanks.